### PR TITLE
[XTG-323] Edit and Delete Leaderboard and Achievement

### PR DESCRIPTION
**Pull request by ramirlm**
This PR adds two modals on Edit Game Page, for inserting/updating both Achievements and Leaderboards.

Also fixes the layout of the page when inserting a new Game, which was weird. 

This is how it's
Captura de Tela 2022-06-24 às 18 25 10
Captura de Tela 2022-06-24 às 18 25 21
 looking like:

### CODE REVIEW

**Comment by ramirlm** [comment]
Forgot to add date-fns to format the dates. Will do it soon

File: src/ui/AddOrEditLeaderboardModal/AddOrEditLeaderboardModal.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                            <TextInput
+                            label="Name"
+                            {...getFieldProps("name")}
+                            {...getFieldMeta("name")}
+                            />
+                        </div>
+
+                        <div>
+                            <Dropdown fieldProps={getFieldProps("scoreStrategy")} label="Score Strategy">
+                                <option value="highest" label="Highest" />
```
Comment (ramirlm) [inline src/ui/AddOrEditLeaderboardModal/AddOrEditLeaderboardModal.tsx]: Will add those values to a variable

File: src/ui/AddOrEditLeaderboardModal/AddOrEditLeaderboardModal.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                            <Dropdown fieldProps={getFieldProps("resetStrategy")} label="Reset Strategy">
+                                <option value="daily" label="Daily" />
+                                <option value="weekly" label="Weekly" />
+                                <option value="monthly" label="Monthly" />
+                                <option value="never" label="Never" />
+                            </Dropdown>
+                        </div>
+                        <div className="mt-8 flex justify-center flex-1">
+                            <Button type="submit" disabled={!isValid} >
+                                Save
```
Comment (ramirlm) [inline src/ui/AddOrEditLeaderboardModal/AddOrEditLeaderboardModal.tsx]: isValid behavior is a bit weird, fixing it

**Review by ccmoralesj — COMMENTED** [review]
This looks great. One question. Is this in sync with the new API endpoints structure? I think it is, but please talk with @fredericobraga85 to be up to date.

**Comment by ramirlm** [comment]
> This looks great. One question. Is this in sync with the new API endpoints structure? I think it is, but please talk with @fredericobraga85 to be up to date.

Yeah it's already updated with latest API endpoints, @ccmoralesj thanksss :)

**Pull request merged**